contracts/governance/Governor.sol

```diff
@@ -760,68 +761,24 @@ abstract contract Governor is Context, ERC165, EIP712, Nonces, IGovernor, IERC72
-        uint256 len = bytes(description).length;
+        uint256 length = bytes(description).length;
 
         // Length is too short to contain a valid proposer suffix
-        if (len < 52) {
+        if (length < 52) {
             return true;
         }
 
-        // Extract what would be the `#proposer=0x` marker beginning the suffix
-        bytes12 marker;
-        assembly ("memory-safe") {
-            // - Start of the string contents in memory = description + 32
-            // - First character of the marker = len - 52
-            //   - Length of "#proposer=0x0000000000000000000000000000000000000000" = 52
-            // - We read the memory word starting at the first character of the marker:
-            //   - (description + 32) + (len - 52) = description + (len - 20)
-            // - Note: Solidity will ignore anything past the first 12 bytes
-            marker := mload(add(description, sub(len, 20)))
-        }
+        // Extract what would be the `#proposer=` marker beginning the suffix
+        bytes10 marker = bytes10(_unsafeReadBytesOffset(bytes(description), length - 52));
 
         // If the marker is not found, there is no proposer suffix to check
-        if (marker != bytes12("#proposer=0x")) {
+        if (marker != bytes10("#proposer=")) {
             return true;
         }
 
-        // Parse the 40 characters following the marker as uint160
-        uint160 recovered = 0;
-        for (uint256 i = len - 40; i < len; ++i) {
-            (bool isHex, uint8 value) = _tryHexToUint(bytes(description)[i]);
-            // If any of the characters is not a hex digit, ignore the suffix entirely
-            if (!isHex) {
-                return true;
-            }
-            recovered = (recovered << 4) | value;
-        }
-
-        return recovered == uint160(proposer);
-    }
-
-    /**
-     * @dev Try to parse a character from a string as a hex value. Returns `(true, value)` if the char is in
-     * `[0-9a-fA-F]` and `(false, 0)` otherwise. Value is guaranteed to be in the range `0 <= value < 16`
-     */
-    function _tryHexToUint(bytes1 char) private pure returns (bool isHex, uint8 value) {
-        uint8 c = uint8(char);
-        unchecked {
-            // Case 0-9
-            if (47 < c && c < 58) {
-                return (true, c - 48);
-            }
-            // Case A-F
-            else if (64 < c && c < 71) {
-                return (true, c - 55);
-            }
-            // Case a-f
-            else if (96 < c && c < 103) {
-                return (true, c - 87);
-            }
-            // Else: not a hex char
-            else {
-                return (false, 0);
-            }
-        }
+        // Check that the last 42 characters (after the marker) is a properly formatted address.
+        (bool success, address recovered) = Strings.tryParseAddress(description, length - 42, length);
+        return !success || recovered == proposer;
```

Comment:
Sholdn't this be?

```suggestion
        return success && recovered == proposer;
```

Reply:
If the address is not properly formated, then we don't consider it and the proposal should pass. At least that is the current behavior.

For example if you have a description that ends in

```#proposer=NULL&something=xxxxxxxxxxxxxxxxxxxxxxxxxxx```
The "#proposer=" marker is at the right location, but we should not apply any restrictions

Reply:
For ref: https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5166/files#diff-66653f6d39b7c3e38c356d782e05f4871f34553eda930c07a6286344eb44d2e1L791